e2e/fetchResponses.ts

```diff
@@ -1,18 +1,29 @@
-import { writeFile } from 'fs/promises';
+/* eslint-disable @typescript-eslint/no-var-requires */
 
-import { Hex, createClient, http, sha256 } from 'viem';
-import { getBlockNumber } from 'viem/actions';
+// this file is ran locally with ts-node to fetch the responses from the swap quotes urls
+require('dotenv').config();
+const { writeFile } = require('fs/promises');
 
-import urls from './mocks/mock_swap_quotes_urls.json';
+const { createClient, http, sha256 } = require('viem');
+const { getBlockNumber } = require('viem/actions');
 
-const client = createClient({ transport: http(process.env.ETH_MAINNET_RPC) });
-const blockNumber = await getBlockNumber(client);
-console.log(blockNumber);
+const urls = require('./mocks/mock_swap_quotes_urls.json');
 
-await Promise.all(
-  urls.map(async (url) => {
-    const res = await fetch(url).then((r) => r.text());
-    const hash = sha256(url as Hex);
-    writeFile(`./mocks/swap_quotes/${hash}.json`, res);
-  }),
-);
+(async () => {
+  const client = createClient({ transport: http(process.env.ETH_MAINNET_RPC) });
+  const blockNumber = await getBlockNumber(client);
+
+  // check initial blockNumber
+  console.log('OUTSIDE FUNCTION', blockNumber.toString());
+
+  await Promise.all(
+    urls.map(async (url: RequestInfo | URL) => {
+      const res = await fetch(url).then((r) => r.text());
+      const hash = sha256(url);
+
+      // check if blockNumber is the same the entire time
+      console.log('INSIDE FUNCTION', blockNumber.toString());
```

Comment:
your not fetching again so this is gonna always be the same

Reply:
🔫